src/ripper/ripcddialog.cpp

```diff
@@ -53,7 +53,8 @@ RipCDDialog::RipCDDialog(QWidget* parent)
     : QDialog(parent),
       ui_(new Ui_RipCDDialog),
       ripper_(new Ripper(this)),
-      working_(false) {
+      working_(false),
+      loader_(new CddaSongLoader) {
```

Comment:
`loader_` needs to be a `unique_ptr` otherwise this will leak.

Reply:
How about just setting the parent and let Qt handle it, i.e. 
` loader_(new CddaSongLoader(QUrl(), this))`?

Reply:
Either is fine

Reply:
Done